src/app/routes/cpsAsset/withFrontPageData/index.js

```diff
@@ -0,0 +1,35 @@
+import { pathOr } from 'ramda';
+
+const extractGroupItems = type => frontPageData => {
+  const featuresGroup = pathOr([], ['content', 'groups'], frontPageData).find(
+    group => group.type === type,
+  );
+  return pathOr([], ['items'], featuresGroup);
+};
+
+const extractFeaturesData = extractGroupItems('feature-main');
+const extractTopStoriesData = extractGroupItems('responsive-top-stories'); // added as an example of retrieving for top-stories
+
+const withFrontPageData = getInitialData => async routePath => {
+  const [, service] = routePath.split('/');
```

Comment:
This is simplistic, and does not cover service variants.

If this POC is accepted, it'll have to be expanded.